packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -474,13 +475,28 @@ export class Table extends Resource {
   }
 
   /**
-   * Permis an IAM principal all stream data read operations for this
+   * Permits an IAM Principal to list all DynamoDB Streams.
+   * @param grantee The principal (no-op if undefined)
+   */
+  public grantTableListStreams(grantee: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions: ['dynamodb:ListStreams'],
+      resourceArns: [
+        Lazy.stringValue({ produce: () => this.tableStreamArn ? `${this.tableArn}/stream/*` : Aws.NO_VALUE })
```

Comment:
I don't understand this condition... If `tableStreamArn` is defined, then I think it should be used here? And if it's not, then I think it should be: `` `${this.tableArn}/stream/*` ``? Also, I don't think `Lazy` is needed here, unless I'm missing something (things like `this.tableArn` are already tokens). So, in summary:

```ts
      resourceArns: [
        this.tableStreamArn ? this.tableStreamArn : `${this.tableArn}/stream/*`,
      ],
```

Reply:
oh this to make sure that table is having stream enabled. The method `grantStream` is also using `tableStreamArn` as condition.

Reply:
Right. Except `grantStream` throws an error if it's undefined, which is what I would expect. This instead will fail at CFN deploy time, because `resource` is required in a Policy Statement.

Can you change the behavior to throw if `this.tableStreamArn` is undefined (+ unit test verifying it)?